packages/kokkos-kernels/src/impl/tpls/KokkosSparse_spmv_tpl_spec_decl.hpp

```diff
@@ -91,36 +91,58 @@ namespace Impl {
 
     /* Check that cusparse can handle the types of the input Kokkos::CrsMatrix */
     cusparseIndexType_t myCusparseOffsetType;
-    if(std::is_same<offset_type, int>::value)     {myCusparseOffsetType = CUSPARSE_INDEX_32I;}
-    if(std::is_same<offset_type, int64_t>::value) {myCusparseOffsetType = CUSPARSE_INDEX_64I;}
+    if(std::is_same<offset_type, int>::value)
+      myCusparseOffsetType = CUSPARSE_INDEX_32I;
+    else if(std::is_same<offset_type, int64_t>::value || std::is_same<offset_type, size_t>::value)
+      myCusparseOffsetType = CUSPARSE_INDEX_64I;
+    else 
+      throw std::logic_error("Offset type of CrsMatrix isn't supported by cuSPARSE, yet TPL layer says it is");
     cusparseIndexType_t myCusparseEntryType;
-    if(std::is_same<entry_type, int>::value)      {myCusparseEntryType = CUSPARSE_INDEX_32I;}
-    if(std::is_same<entry_type, int64_t>::value)  {myCusparseEntryType = CUSPARSE_INDEX_64I;}
+    if(std::is_same<entry_type, int>::value)
+      myCusparseEntryType = CUSPARSE_INDEX_32I;
+    else if(std::is_same<entry_type, int64_t>::value)
```

Comment:
Just a quick question, how does this statement work for people using `long` and `long long` types?

Reply:
@lucbv It is a bit annoying for users but KokkosKernels ETI and TPL layer only supports ``int64_t`` specifically, which is typedef'd to either ``long`` or ``long long`` depending on the platform.